spec/index.bs

```diff
@@ -726,8 +714,49 @@ interface GPUBufferUsage {
 };
 </script>
 
+In order to guarantee immediate, efficient, and portable mapping, buffers
+marked for {{GPUBufferUsage/MAP_WRITE}} are only writable by the CPU,
+and cannot be also marked for {{GPUBufferUsage/COPY_DST}} or {{GPUBufferUsage/STORAGE}}.
+
 ## Buffer Mapping ## {#buffer-mapping}
 
+<script type=idl>
+typedef unsigned long GPUMapFlags;
+interface GPUMap {
+    const GPUMapFlags READ  = 0x0001;
+    const GPUMapFlags WRITE = 0x0002;
+};
+</script>
+
+{{GPUBuffer/map()}} returns null if the device is lost.
+
+Because MAP_WRITE is writable only by the CPU, the contents of the ArrayBuffer
+are unchanged between unmap and map. (Or, if the buffer has not been previously
+mapped, the initial buffer contents.
+
+If the contents of a mapping without {{GPUMap/WRITE}} are changed, on unmap(),
+the device is lost.
+Additionally, writing to a read-only ArrayBuffer returned from mapping without
+{{GPUMap/WRITE}} MAY throw an error.
+
+Only one mapping may be active per buffer at a time.
+
+With {{GPUQueue/createFence()}} and awaiting {{GPUFence/onCompletion()}}, it's possible
```

Comment:
It sounds like the idea is that, before reading from anything, the Javascript is supposed to wait on a fence to make sure the thing it's reading from has already completed. And if the JS doesn't wait on fence itself, then the runtime will wait on a fence on its behalf, inside the `map()` call. Is that accurate?

And for writing, the JS is supposed to wait on a fence to know that the thing it's writing to isn't being used by the GPU, and if it is, the implementation will make a copy under the hood inside the `unmap()` call. Is that accurate?

If that's accurate, doesn't `unmap()` need to be associated with a queue? So the implementation can issue the copy on that queue, and so content authors know exactly when their uploaded writes will be visible by the device.